lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1520,7 +1520,11 @@ namespace Js
         ScriptContext* scriptContext = instance->GetScriptContext();
         DynamicType* cachedDynamicType = nullptr;
         BOOL isJsrtExternalType = instance->GetType()->IsJsrtExternal();
-        DynamicType* oldType = isJsrtExternalType ? 0 : instance->GetDynamicType();
+        // Share the type of all JsrtExternal objects which will have 'newPrototype' as their new prototype.
+        // TypeTransitionMap is a map with key as weakReference and to store it, key need to have a valid heap block.
+        // So instead of using some constant, use newPrototype as the key for saving newType for such objects
+        // in TypeTransitionMap.
+        DynamicType* oldType = isJsrtExternalType ? reinterpret_cast<DynamicType*>(newPrototype) : instance->GetDynamicType();
```

Comment:
I had this concern when I initially implemented it but I didn't find a reason to have separate types for scenarios I was optimizing it for. I will revisit this and see if I can come up with better fix, if needed. Note that this PR doesn't change the original design though except that the key that was being used for `JsrtExternalObject` has been changed  from `0`  to `newPrototype`.